specifyweb/frontend/js_src/lib/components/WorkBench/WbSpreadsheet.tsx

```diff
@@ -298,144 +280,190 @@ function WbSpreadsheetComponent({
             disambiguate: {
               name: wbText.disambiguate(),
               disabled: (): boolean =>
-                // TODO: Figure out if these commented lines can be removed
-                // typeof this.uploadedView === 'function' ||
-                // typeof this.coordinateConverterView === 'function' ||
-                !disambiguation.isAmbiguousCell() ||
-                !hasPermission('/workbench/dataset', 'update'),
+                !disambiguation.isAmbiguousCell() || isReadOnly,
               callback: () => openDisambiguationDialog(),
             },
             separator_1: '---------',
-            fill_down: fillCellsContextMenuItem(hotRef, 'down'),
-            fill_up: fillCellsContextMenuItem(hotRef, 'up'),
+            fill_down: fillCellsContextMenuItem(hot!, 'down', isReadOnly),
```

Comment:
don't lie to typescript - that could cause runtime exceptions if that function would try to access `hot` while it's not defined

would handsontable let you do this instead?

```suggestion
            fill_down: hot === undefined ? undefined : fillCellsContextMenuItem(hot, 'down', isReadOnly),
```

Reply:
Throws a type error. I can do this instead:
```
const contextMenuConfig: DetailedSettings | undefined = 
  hot === undefined ? undefined : { /* rest of the config code */ };
```
What do you think?

Reply:
sounds good!